waterbox/common.mak

```diff
@@ -0,0 +1,104 @@
+# common parts of all waterbox cores
+
+WATERBOX_DIR := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+ROOT_DIR := $(shell dirname $(realpath $(lastword $(filter-out $(lastword $(MAKEFILE_LIST)), $(MAKEFILE_LIST)))))
+OUTPUTDLL_DIR := $(realpath $(WATERBOX_DIR)/../output/dll)
+OBJ_DIR := $(ROOT_DIR)/obj/release
+DOBJ_DIR := $(ROOT_DIR)/obj/debug
+EMULIBC_OBJS := $(WATERBOX_DIR)/emulibc/obj/release/emulibc.c.o
+EMULIBC_DOBJS := $(WATERBOX_DIR)/emulibc/obj/debug/emulibc.c.o
+SYSROOT := $(WATERBOX_DIR)/sysroot
+ifdef NEED_LIBCO
+EMULIBC_OBJS := $(EMULIBC_OBJS) $(shell find $(WATERBOX_DIR)/libco/obj/release -type f -name '*.o')
+EMULIBC_DOBJS := $(EMULIBC_DOBJS) $(shell find $(WATERBOX_DIR)/libco/obj/debug -type f -name '*.o')
+endif
+
+print-%: ;
+	@echo $* = $($*)
+
+#LD_PLUGIN := $(shell gcc --print-file-name=liblto_plugin.so)
+
+CC := $(SYSROOT)/bin/musl-gcc
+CCFLAGS := $(CCFLAGS) -mabi=ms -fvisibility=hidden -I$(WATERBOX_DIR)/emulibc -Wall -mcmodel=large \
+	-mstack-protector-guard=global
+LDFLAGS := $(LDFLAGS) -fuse-ld=gold -static -Wl,-Ttext,0x0000036f00000000 #-Wl,--plugin,$(LD_PLUGIN)
+CCFLAGS_DEBUG := -O0 -g
+CCFLAGS_RELEASE := -O3 -flto
+LDFLAGS_DEBUG :=
+LDFLAGS_RELEASE :=
+CXXFLAGS := $(CXXFLAGS) -I$(SYSROOT)/include/c++/v1 -fno-use-cxa-atexit
+CXXFLAGS_DEBUG :=
+CXXFLAGS_RELEASE :=
+
+EXTRA_LIBS := -L $(SYSROOT)/lib/linux -lclang_rt.builtins-x86_64
+ifneq ($(filter %.cpp, $(SRCS)), )
+EXTRA_LIBS := -lc++ -lc++abi -lunwind $(EXTRA_LIBS)
+endif
+
+_OBJS := $(addsuffix .o, $(realpath $(SRCS)))
+OBJS := $(patsubst $(ROOT_DIR)%, $(OBJ_DIR)%, $(_OBJS))
+DOBJS := $(patsubst $(ROOT_DIR)%, $(DOBJ_DIR)%, $(_OBJS))
+
+$(OBJ_DIR)/%.c.o: %.c
+	@echo cc $<
+	@mkdir -p $(@D)
+	@$(CC) -c -o $@ $< $(CCFLAGS) $(CCFLAGS_RELEASE)
+$(OBJ_DIR)/%.cpp.o: %.cpp
+	@echo cxx $<
+	@mkdir -p $(@D)
+	@$(CC) -c -o $@ $< $(CCFLAGS) $(CXXFLAGS) $(CCFLAGS_RELEASE) 
```

Comment:
At least one of our cores needs CCFLAGS that aren't CXXFLAGS and currently spews 9 billion errors as a result.  Revisit